src/data/list/perm.lean

```diff
@@ -1395,3 +1407,16 @@ end
 end permutations
 
 end list
+
+open list
+
+/-- The list obtained from a permutation of a tuple `f` is permutation equivalent to
+the list obtained from `f`. -/
+lemma equiv.perm.of_fn_comp_perm {n : ℕ} {α : Type uu} (σ : equiv.perm (fin n)) (f : fin n → α) :
+  of_fn (f ∘ σ) ~ of_fn f :=
+begin
+  rw [of_fn_eq_map, of_fn_eq_map, ←map_map],
+  apply perm.map_congr,
+  rw [perm_ext ((nodup_fin_range n).map σ.injective) $ nodup_fin_range n],
+  simpa only [mem_map, mem_fin_range, true_and, iff_true] using σ.surjective
```

Comment:
I think this is worth extracting to another lemma,
```lean
lemma equiv.perm.map_fin_range_perm {n : ℕ} (σ : equiv.perm (fin n)) :
  map σ (fin_range n) ~ fin_range n :=
```

Reply:
OK, that might be useful.

Reply:
This can be generalized to [a result about `fintype`](https://github.com/leanprover-community/mathlib/compare/well_founded_induction_bot...alreadydone:mathlib:well_founded_induction_bot?expand=1#diff-b87979ba9782630fd648bc6c1a706c121e8e03df0a76ac2c8b36633b3a10149bR1736-R1738) and not just `fin n`; however the proof must be moved to a later file since the current file doesn't even know about `multiset`. The fintype result can be strengthened to an [iff_bijective result](https://github.com/leanprover-community/mathlib/compare/well_founded_induction_bot...alreadydone:mathlib:well_founded_induction_bot?expand=1#diff-b87979ba9782630fd648bc6c1a706c121e8e03df0a76ac2c8b36633b3a10149bR1710-R1711); one direction is a bit complicated but the direction we want in this PR is easy from [finset.map_perm](https://leanprover-community.github.io/mathlib_docs/data/finset/basic.html#finset.map_perm).

Maybe these are worth another PR?

Reply:
Thanks for the improvements!

Maybe the best way to proceed is for you to take over and PR your version (and we close thisPR)? Or maybe you can adopt this PR and commit your changes? (I don't know yet how the workflow is in such cases...)

Reply:
I pushed the branch to my fork instead of the official repo because I think you (and reviewers) may not agree with all changes I made over there. Whatever changes you agree with, you can feel free to merge them, or apply them manually if you prefer. (Since my branch is based on this branch, no conflict would need to be resolved from merging.) I don't think it's a good idea to close this PR, as my changes were inspired by comments here, and it's nice to keep comments in one place when possible.

Reply:
OK; I'll have a careful look and make the changes in the next few days.